helper/certutil/types.go

```diff
@@ -202,17 +220,19 @@ func (p *ParsedCertBundle) ToCertBundle() (*CertBundle, error) {
 	}
 
 	if p.PrivateKeyBytes != nil && len(p.PrivateKeyBytes) > 0 {
+		block.Type = string(p.PrivateKeyFormat)
 		block.Bytes = p.PrivateKeyBytes
-		switch p.PrivateKeyType {
-		case RSAPrivateKey:
-			result.PrivateKeyType = "rsa"
-			block.Type = "RSA PRIVATE KEY"
-		case ECPrivateKey:
-			result.PrivateKeyType = "ec"
-			block.Type = "EC PRIVATE KEY"
-		default:
-			return nil, InternalError{"Could not determine private key type when creating block"}
+
+		//Handle bundle not parsed by us
+		if block.Type == "" {
+			switch p.PrivateKeyType {
+			case ECPrivateKey:
```

Comment:
Alright, sounds like a plan. I've got another thought for a minor tweak. What would you think about changing the PrivateKeyType to a string rather than int?

```go
type PrivateKeyType string

const (
	UnknownPrivateKey PrivateKeyType = ""
	RSAPrivateKey     PrivateKeyType = "rsa"
	ECPrivateKey      PrivateKeyType = "ec"
)
```

Then we can just toss the same thing around for parsed and raw cert bundles, and instead of a switch it can be a direct assignment. Since the strings are 3 chars max, I think it would still only take up a single machine word (not sure on 32 bit arch., though)